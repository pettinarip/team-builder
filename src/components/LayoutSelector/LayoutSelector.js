import React from 'react'
import PropTypes from 'prop-types'

const LayoutSelector = ({ layouts, selected, onLayoutClick }) => {
  return (
    <div className='layout-selector list-container'>
      <div className='list'>
        {layouts.map(layout =>
          <div
            className={`item ${selected && selected.id === layout.id ? 'selected' : ''}`}
            key={layout.id}
          >
            <div
              className='layout-description'
              onClick={() => onLayoutClick(layout.id)}
            >
              {layout.config.join('-')}
            </div>
          </div>
        )}
      </div>
    </div>
  )
}

LayoutSelector.propTypes = {
  layouts: PropTypes.array.isRequired,
  selected: PropTypes.object,
  onLayoutClick: PropTypes.func.isRequired
}

export default LayoutSelector
