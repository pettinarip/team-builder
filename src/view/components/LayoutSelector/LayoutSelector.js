import React from 'react'
import PropTypes from 'prop-types'

const LayoutSelector = ({ layouts, selected, onLayoutClick }) => {
  return (
    <div className='layout-selector list-container'>
      <div className='list'>
        {layouts.map(layout =>
          <div
            className={
              selected && selected.id === layout.id
              ? 'item selected'
              : 'item'
            }
            key={layout.id}
          >
            <div
              className='layout-description'
              onClick={() => onLayoutClick(layout.id)}
            >
              {layout.name}
            </div>
          </div>
        )}
      </div>
    </div>
  )
}

LayoutSelector.propTypes = {
  layouts: PropTypes.array.isRequired,
  selected: PropTypes.object,
  onLayoutClick: PropTypes.func.isRequired
}

export default LayoutSelector
