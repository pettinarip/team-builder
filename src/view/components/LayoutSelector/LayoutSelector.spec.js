import React from 'react'
import { shallow } from 'enzyme'
import LayoutSelector from './LayoutSelector'

describe('LayoutSelector component', () => {
  it('should render the component properly', () => {
    const component = mountLayoutSelector({
      layouts: [{
        id: 1,
        name: '4-4-2',
        config: [
          { x: 10, y: 10 },
          { x: 10, y: 10 },
          { x: 10, y: 10 }
        ]
      }, {
        id: 2,
        name: '3-5-2',
        config: [
          { x: 10, y: 10 },
          { x: 10, y: 10 },
          { x: 10, y: 10 }
        ]
      }]
    })

    expect(component).toMatchSnapshot()
  })

  it('should call onLayoutClick function when a layout is clicked', () => {
    const layoutClick = jest.fn()
    const component = mountLayoutSelector({
      layouts: [{
        id: 1,
        name: '4-4-2',
        config: [
          { x: 10, y: 10 },
          { x: 10, y: 10 },
          { x: 10, y: 10 }
        ]
      }],
      onLayoutClick: layoutClick
    })

    component.find('.layout-description').simulate('click')
    expect(layoutClick).toHaveBeenCalled()
  })

  it('should set selected class when a selected layout is passed', () => {
    const component = mountLayoutSelector({
      layouts: [{
        id: 1,
        name: '4-4-2',
        config: [
          { x: 10, y: 10 },
          { x: 10, y: 10 },
          { x: 10, y: 10 }
        ]
      }, {
        id: 2,
        name: '3-5-2',
        config: [
          { x: 10, y: 10 },
          { x: 10, y: 10 },
          { x: 10, y: 10 }
        ]
      }],
      selected: {
        id: 1,
        name: '4-4-2',
        config: [
          { x: 10, y: 10 },
          { x: 10, y: 10 },
          { x: 10, y: 10 }
        ]
      }
    })

    expect(component).toMatchSnapshot()
  })
})

function mountLayoutSelector (props) {
  const propsToUse = {
    onLayoutClick () {},
    layouts: [],
    ...props
  }

  return shallow(<LayoutSelector {...propsToUse} />)
}
