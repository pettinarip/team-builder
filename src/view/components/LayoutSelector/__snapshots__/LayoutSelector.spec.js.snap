// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutSelector component should render the component properly 1`] = `
<div
  className="layout-selector list-container"
>
  <div
    className="list"
  >
    <div
      className="item"
    >
      <div
        className="layout-description"
        onClick={[Function]}
      >
        4-4-2
      </div>
    </div>
    <div
      className="item"
    >
      <div
        className="layout-description"
        onClick={[Function]}
      >
        3-5-2
      </div>
    </div>
  </div>
</div>
`;

exports[`LayoutSelector component should set selected class when a selected layout is passed 1`] = `
<div
  className="layout-selector list-container"
>
  <div
    className="list"
  >
    <div
      className="item selected"
    >
      <div
        className="layout-description"
        onClick={[Function]}
      >
        4-4-2
      </div>
    </div>
    <div
      className="item"
    >
      <div
        className="layout-description"
        onClick={[Function]}
      >
        3-5-2
      </div>
    </div>
  </div>
</div>
`;
